fix: payment builder table tablet layout

joanna-pagepro opened the pull request:
## Description

- Payment Builder table tablet view is the same as the mobile view

A bug with not showing the total values in the table was resolved in this PR: https://github.com/JoinColony/colonyCDapp/pull/2102

## Testing

* Step 1. Open Action Sidebar on tablet screen size i.e. between 1040px and 480px.
* Step 2. Select "Payment builder" action type
* Step 3. Add payment

## Diffs

**New stuff** ✨

* no new stuff

**Changes** 🏗

* additional `changeLayoutBreakpoint` prop added to the `Table` component so that we can pick if the mobile view should start from tablet or mobile breakpoint.

**Deletions** ⚰️

* no deletions


Resolves https://github.com/JoinColony/colonyCDapp/issues/2072

bassgeta commented on src/components/v5/common/Table/Table.tsx:
```diff
@@ -375,18 +386,42 @@ const Table = <T,>({
                 'sm:grid-cols-[1fr_auto_auto]':
                   canGoToNextPage ||
-                  (additionalPaginationButtonsContent && isMobile),
+                  (additionalPaginationButtonsContent &&
```
This component already has way too many booleans, but should we wrap all of these into variables? it's getting a bit unreadable

bassgeta reviewed (changes requested):
Works as it should, so nice job with this! But the `Table` component is getting unreadable with this many properties and conditional rendering.
1. Do we ever want to display a table horizontally on a tablet and vertically on mobile? If not, then I think we should just make the necessary tables vertical. Dunno what the design is, but we could probably work with having mobile and tablet components the same.
2. Should we break up the `Table` component to have pagination as a separate component?

joanna-pagepro commented:
@bassgeta I moved the pagination component to a separate file and updated the props a bit so that we won't have to check the breakpoint to change the layout in the component. I think that it's more readable now.

rdig reviewed (changes requested):
When I'm on a smaller screen size (~700px in this case), I cannot open the "Decision Method" dropdown

I've recorded the interaction below

[Screencast from 2024-03-30 17-27-57.webm](https://github.com/JoinColony/colonyCDapp/assets/1193222/4534cbce-3ea0-40cb-9671-c9f59c8579d5)

joanna-pagepro commented:
@rdig I think that we have a separate issue added to fix that. I found one with the action sidebar: https://github.com/JoinColony/colonyCDapp/issues/2078. The problem is global

bassgeta reviewed (approved):
Nice job, the table feels lighter now :+1:

iamsamgibbs reviewed (changes requested):
Great work on this so far, only minor pick up from me is the button to open this menu I think should be aligned right, and the menu shouldn't span the entire width.

<img width="961" alt="Screenshot 2024-04-04 at 10 57 04" src="https://github.com/JoinColony/colonyCDapp/assets/34915414/1910fc5a-f3f3-454f-841c-bc646674aabd">

joanna-pagepro commented:
@iamsamgibbs should be ok now

iamsamgibbs reviewed (approved):
Thanks for the change, all looks good now!

On tablet:
<img width="680" alt="Screenshot 2024-04-05 at 13 45 52" src="https://github.com/JoinColony/colonyCDapp/assets/34915414/f77d1813-7a6a-490c-8e91-4f0eeadea555">

And on mobile:
<img width="666" alt="Screenshot 2024-04-05 at 13 46 01" src="https://github.com/JoinColony/colonyCDapp/assets/34915414/52724b7d-cfcd-4601-82a0-07b74e5bfdb8">

joanna-pagepro commented:
@rdig can you take a look?

bassgeta reviewed (approved):
Still looks good, reapproving :+1:
